lib/e2e/src/assert.rs

```diff
@@ -26,3 +30,28 @@ impl<R: SolError> Assert<R> for alloy::contract::Error {
         }
     }
 }
+
+pub trait Emits<E> {
+    /// Asserts that transaction emitted an `expected` event.
+    fn emits(&self, expected: E);
+}
+
+impl<E> Emits<E> for TransactionReceipt
+where
+    E: SolEvent,
+    E: PartialEq,
+    E: std::fmt::Debug,
+{
+    fn emits(&self, expected: E) {
+        // Extract all events the are the expected type;
+        let emitted = self
+            .inner
+            .logs()
+            .iter()
+            .filter_map(|log| log.log_decode().ok())
+            .map(|log| log.inner.data)
+            .any(|event| expected == event);
+
+        assert_eq!(emitted, true, "Event {:?} not emitted", expected);
```

Comment:
Since we are adding an assertion message ("Event ....") maybe it makes sense to just `assert!` here instead of `assert_eq!`? wdyt?

Reply:
Ok.